horreum-web/src/domain/runs/DatasetComparison.tsx

```diff
@@ -213,28 +210,71 @@ type ViewComparisonProps = {
 
 function ViewComparison({headers, view, datasets, alerting}: ViewComparisonProps) {
     const [loading, setLoading] = useState(false)
-    const [rows, setRows] = useState<IRow[]>()
-    const token = useSelector(tokenSelector)
+    const [rows, setRows] = useState<IRow[]>([])
     useEffect(() => {
         setLoading(true)
         Promise.all(datasets.map(ds => datasetApi.getSummary(ds.id, view.id)))
             .then(
                 summaries => {
-                    setRows(
-                        view.components.map(vc => ({
-                            cells: [
-                                vc.headerName,
-                                ...summaries.map(summary => {
-                                    const render = renderValue(
-                                        vc.render,
-                                        vc.labels.length == 1 ? vc.labels[0] : undefined,
-                                        token
-                                    )
-                                    return render(summary.view?.[vc.id], summary)
-                                }),
-                            ],
-                        }))
-                    )
+                    const rows: any[][] = []
+                    summaries.forEach((summary, summary_index) => {
+                        let active_view = summary.view
+                        if (active_view != undefined) {
+                            view.components.forEach((component, component_index) => {
+                                let row = rows.find(r => r[0] === component.headerName)
+                                const pos = component_index + 1
+                                // @ts-ignore
+                                const active_component = active_view[pos]
+                                const ob = JSON.stringify( active_component )
+                                let beginning = null
+                                let val = null
+                                if ("\"" === ob.substring(ob.length -2, ob.length - 1)) {
```

Comment:
Please can you explain why we need to create a Json String from a Number Array and then perform String manipulation on the result? Why can we not manipulate the objects directly?

Reply:
because what you assume to be an object is implemented as [a string type](https://github.com/Hyperfoil/Horreum/blob/master/horreum-api/src/main/java/io/hyperfoil/tools/horreum/api/services/DatasetService.java#L133).
Converting it to an object would be the answer. But accessing the object property returns `undefined`.

Reply:
If we need an object we can call `JSON.parse()` and cast to an any type. calling `JSON.stringify( )` on a string and then performing string manipulation is brittle and will break.

Accessing views by position active_view[pos] that is also fragile and likely to break